lib/src/util/windows/user.cc

```diff
@@ -44,15 +46,31 @@ namespace facter { namespace util { namespace windows { namespace user {
 
     string home_dir()
     {
-        string home, alt;
-        if (environment::get("HOME", home)) {
-            return home;
-        } else if (environment::get("HOMEDRIVE", home) && environment::get("HOMEPATH", alt)) {
-            return home + alt;
-        } else if (environment::get("USERPROFILE", home)) {
-            return home;
+        HANDLE temp_token;
+        if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &temp_token)) {
+            LOG_DEBUG("OpenProcessToken call failed: %1%", system_error());
+            return {};
         }
-        return {};
+        scoped_resource<HANDLE> token(temp_token, CloseHandle);
```

Comment:
Thanks @branan for pointing out the `move` business. I noticed there are other places were we move, e.g. https://github.com/puppetlabs/facter/pull/911/files#diff-374aeb3d1d3ffe6398f65723565a063cR30. May be we fix those in a different commit?

Also, I'm wondering if we should be initializing `temp_token` to `INVALID_HANDLE_VALULE` so there's no chance of `CloseHandle` being called on an uninitialized value. I don't think there's a chance of that happening here, just a general best practice concern.